garak/generators/guardrails.py

```diff
@@ -26,14 +28,18 @@ def __init__(self, name, generations=1):
             ) from e
 
         self.name = name
+        self.generations = generations
+        self._load_config(config_root)
         self.fullname = f"Guardrails {self.name}"
 
         # Currently, we use the model_name as the path to the config
         with redirect_stderr(io.StringIO()) as f:  # quieten the tqdm
             config = RailsConfig.from_path(self.name)
             self.rails = LLMRails(config=config)
 
-        super().__init__(name, generations=generations)
+        super().__init__(
+            self.name, generations=self.generations, config_root=config_root
+        )
```

Comment:
Whenever using `Super().__init__` the implementer will need awareness of any side-effects from the call. The base class implementations so far look to be fairly careful about testing stated before blindly assigning values however that is definitely something that needs careful attention when using composition and extension.

Reply:
I think there is some ambiguity still to be aware of, yet I think it is a reasonable edge case.

`Configurable` ensures it only `loads` the config attributes once, and the `base` class will make this it's first action. However as a `super().__init__()` could call another `super()__init__()` the are possible side effects. In practice `Configurable` will apply a configuration file value over any value not provided by the constructor of the class and will then inject any default parameter values as defined in the original instantiated or inherited from a base class that are not found in the config file. 

This means once applied by instance variables are set calling `self._load_config(config_root)` any changes by subsequent code will override the values from configurable. This is why once `self._load_config` is called I attempted to ensuring that any calls to `super()` are passing in the loaded `self.*` values instead of the argument passed from the original constructor.

As mentioned the existing `base` implementations are good about checking if an attribute has already been initialized before updating them, however that does not alleviate the need to take a close look at impacts from parent initializers. There are many cases where the original instantiated class calls super and the does post processing based on the final initialize local variables.

I acknowledge there could be some gotchas in this approach, I don't currently have a patten I believe to be a superior way to ensure nothing in an intermediate constructor is overwritten by a constructor below it. At some point we might need a linter that looks for patterns that might have these kinds of side effects to help us guide contributors to be aware or avoid such impacts.